client/app/scripts/superdesk/editor/editor.js

```diff
@@ -11,318 +11,485 @@
 
 'use strict';
 
+/**
+ * Generate click event on given target node
+ *
+ * @param {Node} target
+ */
 function click(target) {
     target.dispatchEvent(new MouseEvent('click'));
 }
 
-function EditorService() {
-    this.settings = {spellcheck: true};
-    this.stopEvents = false;
-    this.addEventListeners = addEventListeners;
-    this.removeEventListeners = removeEventListeners;
-    this.elements = [];
+/**
+ * Replace given dom elem with its contents
+ *
+ * It is like jQuery unwrap
+ *
+ * @param {Node} elem
+ */
+function replaceSpan(elem) {
+    var parent = elem.parentNode;
+    while (elem.hasChildNodes()) {
+        parent.insertBefore(elem.childNodes.item(0), elem);
+    }
 
-    this.KEY_CODES = Object.freeze({
-        Y: 89,
-        X: 90
-    });
+    parent.removeChild(elem);
+}
+
+/**
+ * Remove all elements with given className but keep its contents
+ *
+ * @param {Node} elem
+ * @param {string} className
+ * @return {Node}
+ */
+function removeClass(elem, className) {
+    var node = elem.cloneNode(true);
+    var spans = node.getElementsByClassName(className);
+    while (spans.length) {
+        replaceSpan(spans.item(0));
+    }
 
-    var vm = this;
+    node.normalize();
+    return node;
+}
+
+/**
+ * Find text node + offset for given node and offset
+ *
+ * This will find text node within given node that contains character on given offset
+ *
+ * @param {Node} node
+ * @param {numeric} offset
+ * @return {Object} {node: {Node}, offset: {numeric}}
+ */
+function findTextNode(node, offset) {
+    var tree = document.createTreeWalker(node, NodeFilter.SHOW_TEXT);
+    var currentLength;
+    var currentOffset = 0;
+    while (tree.nextNode()) {
+        tree.currentNode.textContent = tree.currentNode.textContent.replace(String.fromCharCode(65279), '');
```

Comment:
It's not clear what character `String.fromCharCode(65279)` represents, I suggest creating an alias for it.